templates/observability-remote-write-proxy.yml

```diff
@@ -0,0 +1,335 @@
+        app.kubernetes.io/name: observability-remote-write-proxy
+        app.kubernetes.io/version: ${OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_TAG}
+      name: observability-remote-write-proxy
+    spec:
+      replicas: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_REPLICAS}}
+      strategy:
+        rollingUpdate:
+          maxSurge: 25%
+          maxUnavailable: 25%
+        type: RollingUpdate
+      selector:
+        matchLabels:
+          app.kubernetes.io/component: prometheus-remote-write-proxy
+          app.kubernetes.io/name: observability-remote-write-proxy
+      template:
+        metadata:
+          labels:
+            app.kubernetes.io/component: prometheus-remote-write-proxy
+            app.kubernetes.io/name: observability-remote-write-proxy
+        spec:
+          securityContext:
+            runAsNonRoot: true
+            seccompProfile:
+              type: RuntimeDefault
+          affinity:
+            # Prefer not co-locating Observability Remote Write Proxy pods in the
+            # same zone. This means that K8s scheduler will try to spread
+            # the different Observability Remote Write Proxy pods among the
+            # available zones (logical failure domains)
+            podAntiAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+              - podAffinityTerm:
+                  labelSelector:
+                    matchExpressions:
+                    - key: app.kubernetes.io/name
+                      operator: In
+                      values:
+                      - observability-remote-write-proxy
+                  topologyKey: topology.kubernetes.io/zone
+                weight: 100
+          containers:
+          - name: observability-remote-write-proxy
+            image: ${OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_NAME}:${OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_TAG}
+            imagePullPolicy: ${OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_PULL_POLICY}
+            ports:
+            - containerPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+              name: http
+            - containerPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+              name: metrics
+            command:
+              - /proxy
+              - --proxy.listen.port=${OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}
+              - --proxy.metrics.port=${OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}
+              - --proxy.forwardUrl=${OBSERVABILITY_REMOTE_WRITE_PROXY_FORWARD_URL}
+              - --oidc.enabled=${OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_ENABLED}
+              - --oidc.filename=${OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_CONFIG_FILEDIR}/${OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_CONFIG_FILENAME}
+              - --token.verification.enabled=${OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_ENABLED}
+              - --token.verification.url=${OBSERVABILITY_REMOTE_WRITE_PROXY_TOKEN_VERIFICATION_URL}
+            resources:
+              requests:
```

Comment:
Configurable resource requests and limits. If desired, we can configure different resource requests and limits depending on the environment.
For now we intentionally decided that:
* We will set both cpu and memory requests (see defaults in the template)
* We will set a limit of memory. We have set a default that we hope it is high enough that the limit is not reached
* We will not set a limit of cpu. We will track usage from the metrics and once we have observed the behavior for a while we will decide an appropriate cpu limit